R/periods.r

```diff
@@ -393,7 +400,12 @@ setMethod("$<-", signature(x = "Period"), function(x, name, value) {
 #' @export
 period <- function(num = NULL, units = "second", ...) {
   nums <- list(...)
-  if (is.character(num)) {
+  if (is.null(num) && missing(...)) {
```

Comment:
I think this change and the entire spaghetti of the if-elses that follows can be reduced to one if-else with the pre-conversion `num <- as.numeric(num)`. Let me try it out. 

BTW, `period(numeric())` fails ATM.